experimental/Schemes/SpaceGerms.jl

```diff
@@ -48,52 +45,266 @@ A space germ ``(X,O_{(X,x)}``, i.e. a ringed space with underlying scheme ``X``
   end
 end
 
+@doc raw"""
+    HypersurfaceGerm{BaseRingType, RingType, SpecType}
+A hypersurface germ ``(X,O_{(X,x)}``, i.e. a ringed space with underlying scheme ``X`` of type `SpecType` and local ring ``O_{(X,x)}`` of type `RingType` over some base ring ``k`` of type `BaseRingType`.
+"""
+@attributes mutable struct HypersurfaceGerm{
+                 BaseRingType<:Ring,
+                 RingType<:Ring,
+                 SpecType<:Spec} <: AbsSpaceGerm{BaseRingType, RingType}
+  X::SpecType
+  f::RingElem
+
+  function HypersurfaceGerm(X::GermAtClosedPoint,f::MPolyLocRingElem; check::Bool=true)
+    base_ring(modulus(OO(X))) == parent(f) || error("baserings do not match")
+    @check begin
+      (ideal(parent(f),[f]) == modulus(OO(X))) || error("given f does not define given X")
+    end
+    return new{typeof(base_ring(X)), typeof(OO(X)), typeof(X)}(X,f)
+  end
+
+## the following is currently unused....
+## as no backend for groebner computations is currently available in this case
+  function HypersurfaceGerm(X::GermAtGeometricPoint, f::MPolyLocRingElem; check::Bool=true)
+    base_ring(modulus(OO(X))) == parent(f) || error("baserings do not match")
+    @check begin
+      (ideal(parent(f),[f]) == modulus(OO(X))) || error("given f does not define given X")
+    end
+    return new{typeof(base_ring(X)), typeof(OO(X)), typeof(X)}(X,f)
+  end
+end
+
+@doc raw"""
+    CompleteIntersectionGerm{BaseRingType, RingType, SpecType}
```

Comment:
```suggestion
    CompleteIntersectionGerm{BaseRingType, RingType, SpecType}

```
and so on -- there should be an empty line between the signature(s) and the description.